main.cpp

```diff
@@ -0,0 +1,28 @@
+#include <iostream>
+#include <iomanip>
+
+int main() {
+
+    const int SHARES_BOUGHT = 1000;
+    const double PURCHASE_PRICE = 45.50;
+    const double COMMISSION_RATE = .02;
+    const int SHARES_SOLD = 1000;
+    const double SALE_PRICE = 56.90;
+    double amount_paid = SHARES_BOUGHT * PURCHASE_PRICE;
```

Comment:
While `const` or immutable variables should always be hoisted to the top of the scope like you have them. You should try and declare your other variables as close to usage as possible.
Consider this sample:
```
double amount_paid = SHARES_BOUGHT * PURCHASE_PRICE;
double first_commission = amount_paid * COMMISSION_RATE;
std::cout << std::fixed << std::setprecision(2) << "Joe bought " << SHARES_BOUGHT << " shares at $" << PURCHASE_PRICE << " for $" << amount_paid << std::endl;
std::cout << "Joe paid 2% commission to his broker for the purchase which amounted to $" << first_commission << std::endl;

double amount_sold = SHARES_SOLD * SALE_PRICE;
double second_commission = amount_sold * COMMISSION_RATE;
std::cout << "Joe sold his " << SHARES_SOLD << " shares at $" << SALE_PRICE << " for $" << amount_sold << std::endl;
std::cout << "Joe paid 2% commission to his broker for the sale which amounted to $" << second_commission << std::endl;
```

This way our context is much tighter. The first group is about purchases and the second is about sales.